### ISSUE TITLE
Problem accessing /quickstart/index.html. Reason: Not Found

### ISSUE THREAD

**Issue by OneBall2020** [osgi/osgi.enroute#122]
Java Version 11.0.6
Maven Version 3.6.0

The Error I am getting is when I click the "quickstart":

HTTP ERROR 404
Problem accessing /quickstart/index.html. Reason:

    Not Found

I completed the auxiliary instruction: 

_If you’re using a Java version higher than 8 to run this tutorial then you’ll need to set the appropriate runee in the examples/quickstart/app/app.bndrun and then resolve the application. For Java 9 use JavaSE-9, for Java 10 use JavaSE-10 and for Java 11 use JavaSE-11. Once you have made this edit issue the command mvn bnd-indexer:index bnd-indexer:index@test-index bnd-resolver:resolve to generate the index, test index, and resolve the bndrun (you should be able to see the changes in the runbundles list afterwards). Once you’ve done this the first time then you can mvn verify to your heart’s content._

And get the following:
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by com.google.inject.internal.cglib.core.$ReflectUtils$1 (file:/usr/share/maven/lib/guice.jar) to method java.lang.ClassLoader.defineClass(java.lang.String,byte[],int,int,java.security.ProtectionDomain)
WARNING: Please consider reporting this to the maintainers of com.google.inject.internal.cglib.core.$ReflectUtils$1
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
[INFO] Scanning for projects...
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Build Order:
[INFO] 
[INFO] quickstart                                                         [pom]
[INFO] rest                                                               [jar]
[INFO] app                                                                [jar]
[WARNING] The POM for biz.aQute.bnd:bnd-run-maven-plugin:jar:4.1.0 is missing, no dependency information available
[WARNING] Failed to retrieve plugin descriptor for biz.aQute.bnd:bnd-run-maven-plugin:4.1.0: Plugin biz.aQute.bnd:bnd-run-maven-plugin:4.1.0 or one of its dependencies could not be resolved: Failure to find biz.aQute.bnd:bnd-run-maven-plugin:jar:4.1.0 in https://repo.maven.apache.org/maven2 was cached in the local repository, resolution will not be reattempted until the update interval of central has elapsed or updates are forced
[WARNING] The POM for biz.aQute.bnd:bnd-run-maven-plugin:jar:4.1.0 is missing, no dependency information available
[WARNING] Failed to retrieve plugin descriptor for biz.aQute.bnd:bnd-run-maven-plugin:4.1.0: Plugin biz.aQute.bnd:bnd-run-maven-plugin:4.1.0 or one of its dependencies could not be resolved: Failure to find biz.aQute.bnd:bnd-run-maven-plugin:jar:4.1.0 in https://repo.maven.apache.org/maven2 was cached in the local repository, resolution will not be reattempted until the update interval of central has elapsed or updates are forced
[WARNING] The POM for biz.aQute.bnd:bnd-run-maven-plugin:jar:4.1.0 is missing, no dependency information available
[WARNING] Failed to retrieve plugin descriptor for biz.aQute.bnd:bnd-run-maven-plugin:4.1.0: Plugin biz.aQute.bnd:bnd-run-maven-plugin:4.1.0 or one of its dependencies could not be resolved: Failure to find biz.aQute.bnd:bnd-run-maven-plugin:jar:4.1.0 in https://repo.maven.apache.org/maven2 was cached in the local repository, resolution will not be reattempted until the update interval of central has elapsed or updates are forced
[INFO] 
[INFO] ----------< org.osgi.enroute.examples.quickstart:quickstart >-----------
[INFO] Building quickstart 0.0.2-SNAPSHOT                                 [1/3]
[INFO] --------------------------------[ pom ]---------------------------------
[WARNING] The POM for biz.aQute.bnd:bnd-run-maven-plugin:jar:4.1.0 is missing, no dependency information available
[WARNING] Failed to retrieve plugin descriptor for biz.aQute.bnd:bnd-run-maven-plugin:4.1.0: Plugin biz.aQute.bnd:bnd-run-maven-plugin:4.1.0 or one of its dependencies could not be resolved: Failure to find biz.aQute.bnd:bnd-run-maven-plugin:jar:4.1.0 in https://repo.maven.apache.org/maven2 was cached in the local repository, resolution will not be reattempted until the update interval of central has elapsed or updates are forced
[WARNING] The POM for biz.aQute.bnd:bnd-run-maven-plugin:jar:4.1.0 is missing, no dependency information available
[WARNING] Failed to retrieve plugin descriptor for biz.aQute.bnd:bnd-run-maven-plugin:4.1.0: Plugin biz.aQute.bnd:bnd-run-maven-plugin:4.1.0 or one of its dependencies could not be resolved: Failure to find biz.aQute.bnd:bnd-run-maven-plugin:jar:4.1.0 in https://repo.maven.apache.org/maven2 was cached in the local repository, resolution will not be reattempted until the update interval of central has elapsed or updates are forced
[WARNING] The POM for biz.aQute.bnd:bnd-run-maven-plugin:jar:4.1.0 is missing, no dependency information available
[WARNING] Failed to retrieve plugin descriptor for biz.aQute.bnd:bnd-run-maven-plugin:4.1.0: Plugin biz.aQute.bnd:bnd-run-maven-plugin:4.1.0 or one of its dependencies could not be resolved: Failure to find biz.aQute.bnd:bnd-run-maven-plugin:jar:4.1.0 in https://repo.maven.apache.org/maven2 was cached in the local repository, resolution will not be reattempted until the update interval of central has elapsed or updates are forced
[INFO] 
[INFO] --- bnd-indexer-maven-plugin:4.1.0:index (default-cli) @ quickstart ---
[INFO] The configured deployment repository OSGi Snapshots has the same id as one of the remote artifact repositories. It is assumed that these repositories are the same.
[INFO] 
[INFO] --- bnd-indexer-maven-plugin:4.1.0:index (test-index) @ quickstart ---
[INFO] The configured deployment repository OSGi Snapshots has the same id as one of the remote artifact repositories. It is assumed that these repositories are the same.
[INFO] 
[INFO] --- bnd-resolver-maven-plugin:4.1.0:resolve (default-cli) @ quickstart ---
[INFO] 
[INFO] -------------< org.osgi.enroute.examples.quickstart:rest >--------------
[INFO] Building rest 0.0.2-SNAPSHOT                                       [2/3]
[INFO] --------------------------------[ jar ]---------------------------------
[WARNING] The POM for biz.aQute.bnd:bnd-run-maven-plugin:jar:4.1.0 is missing, no dependency information available
[WARNING] Failed to retrieve plugin descriptor for biz.aQute.bnd:bnd-run-maven-plugin:4.1.0: Plugin biz.aQute.bnd:bnd-run-maven-plugin:4.1.0 or one of its dependencies could not be resolved: Failure to find biz.aQute.bnd:bnd-run-maven-plugin:jar:4.1.0 in https://repo.maven.apache.org/maven2 was cached in the local repository, resolution will not be reattempted until the update interval of central has elapsed or updates are forced
[WARNING] The POM for biz.aQute.bnd:bnd-run-maven-plugin:jar:4.1.0 is missing, no dependency information available
[WARNING] Failed to retrieve plugin descriptor for biz.aQute.bnd:bnd-run-maven-plugin:4.1.0: Plugin biz.aQute.bnd:bnd-run-maven-plugin:4.1.0 or one of its dependencies could not be resolved: Failure to find biz.aQute.bnd:bnd-run-maven-plugin:jar:4.1.0 in https://repo.maven.apache.org/maven2 was cached in the local repository, resolution will not be reattempted until the update interval of central has elapsed or updates are forced
[WARNING] The POM for biz.aQute.bnd:bnd-run-maven-plugin:jar:4.1.0 is missing, no dependency information available
[WARNING] Failed to retrieve plugin descriptor for biz.aQute.bnd:bnd-run-maven-plugin:4.1.0: Plugin biz.aQute.bnd:bnd-run-maven-plugin:4.1.0 or one of its dependencies could not be resolved: Failure to find biz.aQute.bnd:bnd-run-maven-plugin:jar:4.1.0 in https://repo.maven.apache.org/maven2 was cached in the local repository, resolution will not be reattempted until the update interval of central has elapsed or updates are forced
[INFO] 
[INFO] --- bnd-indexer-maven-plugin:4.1.0:index (default-cli) @ rest ---
[INFO] The configured deployment repository OSGi Snapshots has the same id as one of the remote artifact repositories. It is assumed that these repositories are the same.
[INFO] 
[INFO] --- bnd-indexer-maven-plugin:4.1.0:index (test-index) @ rest ---
[INFO] The configured deployment repository OSGi Snapshots has the same id as one of the remote artifact repositories. It is assumed that these repositories are the same.
[INFO] 
[INFO] --- bnd-resolver-maven-plugin:4.1.0:resolve (default-cli) @ rest ---
[INFO] 
[INFO] --------------< org.osgi.enroute.examples.quickstart:app >--------------
[INFO] Building app 0.0.2-SNAPSHOT                                        [3/3]
[INFO] --------------------------------[ jar ]---------------------------------
[WARNING] The POM for biz.aQute.bnd:bnd-run-maven-plugin:jar:4.1.0 is missing, no dependency information available
[WARNING] Failed to retrieve plugin descriptor for biz.aQute.bnd:bnd-run-maven-plugin:4.1.0: Plugin biz.aQute.bnd:bnd-run-maven-plugin:4.1.0 or one of its dependencies could not be resolved: Failure to find biz.aQute.bnd:bnd-run-maven-plugin:jar:4.1.0 in https://repo.maven.apache.org/maven2 was cached in the local repository, resolution will not be reattempted until the update interval of central has elapsed or updates are forced
[WARNING] The POM for biz.aQute.bnd:bnd-run-maven-plugin:jar:4.1.0 is missing, no dependency information available
[WARNING] Failed to retrieve plugin descriptor for biz.aQute.bnd:bnd-run-maven-plugin:4.1.0: Plugin biz.aQute.bnd:bnd-run-maven-plugin:4.1.0 or one of its dependencies could not be resolved: Failure to find biz.aQute.bnd:bnd-run-maven-plugin:jar:4.1.0 in https://repo.maven.apache.org/maven2 was cached in the local repository, resolution will not be reattempted until the update interval of central has elapsed or updates are forced
[WARNING] The POM for biz.aQute.bnd:bnd-run-maven-plugin:jar:4.1.0 is missing, no dependency information available
[WARNING] Failed to retrieve plugin descriptor for biz.aQute.bnd:bnd-run-maven-plugin:4.1.0: Plugin biz.aQute.bnd:bnd-run-maven-plugin:4.1.0 or one of its dependencies could not be resolved: Failure to find biz.aQute.bnd:bnd-run-maven-plugin:jar:4.1.0 in https://repo.maven.apache.org/maven2 was cached in the local repository, resolution will not be reattempted until the update interval of central has elapsed or updates are forced
[INFO] 
[INFO] --- bnd-indexer-maven-plugin:4.1.0:index (default-cli) @ app ---
[INFO] The configured deployment repository OSGi Snapshots has the same id as one of the remote artifact repositories. It is assumed that these repositories are the same.
[INFO] 
[INFO] --- bnd-indexer-maven-plugin:4.1.0:index (test-index) @ app ---
[INFO] The configured deployment repository OSGi Snapshots has the same id as one of the remote artifact repositories. It is assumed that these repositories are the same.
[INFO] 
[INFO] --- bnd-resolver-maven-plugin:4.1.0:resolve (default-cli) @ app ---
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for quickstart 0.0.2-SNAPSHOT:
[INFO] 
[INFO] quickstart ......................................... SUCCESS [  0.454 s]
[INFO] rest ............................................... SUCCESS [  1.150 s]
[INFO] app ................................................ SUCCESS [  5.360 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  7.771 s
[INFO] Finished at: 2020-01-30T11:25:57-08:00
[INFO] ------------------------------------------------------------------------

Then $mvn verify
Then $java -jar app/target/app.jar

Clicking the "quickstart" returns the 404 ERROR
Checking localhost:8080 opens the Aries JAX-RS Whiteboard.

**Comment by timothyjward** [comment]
Hello, and thank you for reporting this issue. There are two parts to the problem.

1. You've found a hole in the documentation - there's supposed to be a warning box (similar to the one on this page https://enroute.osgi.org/Examples/) telling you that it's a good idea to checkout the release tag rather than using snapshots.
2. There's a bug that crept into the Aries JAX-RS Whiteboard which can (in a lovely timing related way) result in static resources being hidden. This bug has been fixed, but we have been waiting on a release containing the fix to merge into the development branch.

Hopefully by checking out the `7.0.0` tag and repeating you will see the tutorial work correctly. Please let us know if you don't.

**Comment by OneBall2020** [comment]
@timothyjward 
Checking out Version 7.0.0 worked.  Thank you!

**Comment by orbruno** [comment]
@timothyjward 
I am experiencing the same problem, I tried to checkout version 7.0.0 but I get an error message "error: pathspec '7.0.0' did not match any file(s) known to git" I also made sure that the pom.xml file in the example folder would match the recommended changes on the issue "Avoid using the Sonatype OSGi group and use the repositories instead #98"

Thanks!
